### PR TITLE
MNT: Unpin maxversion of Sphinx and Jinja2

### LINKED ISSUE #11725: DOC: Unpin max version of Sphinx when Sphinx 4.0.2 is out (and Jinja2)
**Update:** If you want to build docs in Python 3.10 or newer, see the pins in  #13587 .

Revert #11724 when sphinx-doc/sphinx#9210 is resolved. @saimn tracked it down to sphinx-doc/sphinx#8935 .

https://github.com/astropy/astropy/blob/642d8aea136d274a82928a8735cd25d6905116ed/setup.cfg#L84-L85

Also revert #12988 .

https://github.com/astropy/astropy/blob/f6569ceaeb81afef80d5a6a0a579698f1c27c6cd/setup.cfg#L103

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is another attempt to fix #11725 

Close #12270

If you disagree this needs backporting, please remilestone and remove the backport label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/coordinates/builtin_frames/lsr.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 from .baseradec import BaseRADecFrame, doc_components as doc_components_radec
 from .icrs import ICRS
 from .galactic import Galactic
-from .fk4 import FK4
+from .fk4 import FK4  # noqa
```
Comment (pllim) [inline astropy/coordinates/builtin_frames/lsr.py:17]: @adrn, is this import really necessary?

Comment (adrn) [reply astropy/coordinates/builtin_frames/lsr.py:17]: No, that is vestigial and could be removed now
```suggestion
```

Comment (pllim) [reply astropy/coordinates/builtin_frames/lsr.py:17]: This PR is going nowhere, so I removed it directly in 6c15aa65851dc31e8cac843bd16e83178c8b50bf . Thanks for the response!

**Comment by pllim** [comment]
Whatever magic that worked for photutils is not working here.

**Comment by larrybradley** [comment]
Did you exclude the sub-sub-package from `__all__`?  That's essentially what I did in https://github.com/astropy/photutils/pull/1306 to solve the issue.

**Comment by pllim** [comment]
> Did you exclude the sub-sub-package

What do you mean, @larrybradley ? All I see there is you import `from . import subpackage`. Did I miss something fundamental there?

**Comment by pllim** [comment]
I am not sure if that is possible for coordinates. Magical things seem to happen in that `__init__.py` but I guess I can try again when I get a chance...

**Comment by larrybradley** [comment]
It would be nice to get this fixed (perhaps even upstream in astropy/sphinx-automodapi) because one cannot build the docs locally with Python 3.10 and Sphinx < 4.  That's what motivated me to do the `photutils` workaround.

**Comment by pllim** [comment]
Upstream fix is beyond me. My sphinx-fu is weak...

**Comment by pllim** [comment]
See #13048

**Pull request closed without merging**